processor/core/src/main/java/com/redhat/service/smartevents/processor/GatewayBean.java

```diff
@@ -0,0 +1,11 @@
+package com.redhat.service.smartevents.processor;
+
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+
+public interface GatewayBean<T extends Gateway> {
+    String getType();
+
+    default boolean accept(String gatewayType) {
+        return getType().equals(gatewayType);
```

Comment:
Could we please use `Objects.equals(a, b)` pattern instead of `a.equals(b)`. The first is NullPointerException safe.